# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/ipdl/sync-messages.ini
# Commit: 71e63781b38c
# Full Hash: 71e63781b38cbde024a9057553cae96c98e5ecb2
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-02-13 18:21:33
# Regressor Bug: 1604412
# File Overlap Count: 2
# Description:
#   Bug 1604412 - Enable remote backbuffer GDI compositing r=jmathies,jld
#   
#   This change adds new "remote backbuffer" logic when compositing without
#   HW acceleration on Windows (IE compositing through Cairo using the Win32
#   GDI)
# ==============================================================================

diff -r f2088b5ae50d -r 71e63781b38c ipc/ipdl/sync-messages.ini
--- a/ipc/ipdl/sync-messages.ini	Thu Feb 13 07:13:50 2020 +0200
+++ b/ipc/ipdl/sync-messages.ini	Thu Feb 13 03:59:13 2020 +0000
@@ -1111,18 +1111,21 @@
 description = legacy sync IPC - please add detailed description
 [PCompositorBridge::CheckContentOnlyTDR]
 description = legacy sync IPC - please add detailed description
+[PCompositorWidget::Initialize]
+description = Fallable initialization for the widget. Must be called right after construction before any other messages are sent
+platform = win
 [PCompositorWidget::EnterPresentLock]
-description = legacy sync IPC - please add detailed description
+description = Obtain exclusive access to the widget surface. Used to ensure events don't change the surface while it's being used as a render target
 platform = win
 [PCompositorWidget::LeavePresentLock]
-description = legacy sync IPC - please add detailed description
+description = Release the exclusive lock that EnterPresentLock obtained
 platform = win
 [PCompositorBridge::SupportsAsyncDXGISurface]
 description = legacy sync IPC - please add detailed description
 [PCompositorBridge::PreferredDXGIAdapter]
 description = legacy sync IPC - please add detailed description
 [PCompositorWidget::ClearTransparentWindow]
-description = legacy sync IPC - please add detailed description
+description = Synchronously clear the widget surface. Does not rely on (Enter|Leave)PresentLock. When call returns, window surface has been fully updated with cleared pixel values.
 platform = win
 [PImageBridge::WillClose]
 description = legacy sync IPC - please add detailed description